tests/remotes/ssh.py

```diff
@@ -63,20 +66,73 @@ def get_url(user, port):  # pylint: disable=arguments-differ
         url = f"ssh://{user}@127.0.0.1:{port}{path}"
         return url
 
-    def __init__(self, server):
-        self.server = server
-
-    @cached_property
-    def url(self):
-        return self.get_url(TEST_SSH_USER, self.server.port)
-
     @cached_property
     def config(self):
         return {
             "url": self.url,
             "keyfile": TEST_SSH_KEY_PATH,
         }
 
+    @contextmanager
+    def _ssh(self):
+        from dvc.remote.ssh.connection import SSHConnection
+
+        conn = SSHConnection(
+            host=self.host,
+            port=self.port,
+            username=TEST_SSH_USER,
+            key_filename=TEST_SSH_KEY_PATH,
+        )
+        try:
+            yield conn
+        finally:
+            conn.close()
+
+    def is_file(self):
+        with self._ssh() as _ssh:
```

Comment:
Not very nice, might just create a connection and then add a closing logic to `close`, as it is really intended to be, and then register a fixture finaliser or something.